qiita_db/util.py

```diff
@@ -1366,78 +1368,97 @@ def generate_study_list(study_ids):
-                    AS number_samples_collected,
-                (SELECT array_agg(artifact_id ORDER BY artifact_id)
-                    FROM qiita.study_artifact
-                    LEFT JOIN qiita.artifact USING (artifact_id)
-                    LEFT JOIN qiita.artifact_type USING (artifact_type_id)
-                    WHERE artifact_type='BIOM' AND
-                        study_id = qiita.study.study_id) AS artifact_biom_ids,
-                (SELECT array_agg(row_to_json((publication, is_doi), true))
-                    FROM qiita.study_publication
-                    WHERE study_id=qiita.study.study_id) AS publications,
-                (SELECT array_agg(name ORDER BY email) FROM qiita.study_users
-                    LEFT JOIN qiita.qiita_user USING (email)
-                    WHERE study_id=qiita.study.study_id) AS shared_with_name,
-                (SELECT array_agg(email ORDER BY email) FROM qiita.study_users
-                    LEFT JOIN qiita.qiita_user USING (email)
-                    WHERE study_id=qiita.study.study_id) AS shared_with_email,
-                (SELECT array_agg(study_tag) FROM qiita.per_study_tags
-                    WHERE study_id=qiita.study.study_id) AS study_tags,
-                (SELECT name FROM qiita.qiita_user
-                    WHERE email=qiita.study.email) AS owner
-                FROM qiita.study
-                LEFT JOIN qiita.study_person ON (
-                    study_person_id=principal_investigator_id)
-                WHERE study_id IN %s
-                ORDER BY study_id"""
-        qdb.sql_connection.TRN.add(sql, [tuple(study_ids)])
-        infolist = []
-        for info in qdb.sql_connection.TRN.execute_fetchindex():
-            info = dict(info)
-
-            # publication info
-            info['publication_doi'] = []
-            info['publication_pid'] = []
-            if info['publications'] is not None:
-                for p in info['publications']:
-                    # f1-2 are the default names given
-                    pub = p['f1']
-                    is_doi = p['f2']
-                    if is_doi:
-                        info['publication_doi'].append(pub)
-                    else:
-                        info['publication_pid'].append(pub)
-            del info['publications']
 
-            # pi info
-            info["pi"] = (info['pi_email'], info['pi_name'])
-            del info["pi_email"]
-            del info["pi_name"]
-
-            # shared with
-            info['shared'] = []
-            if info['shared_with_name'] and info['shared_with_email']:
-                for name, email in zip(info['shared_with_name'],
-                                       info['shared_with_email']):
-                    if not name:
-                        name = email
-                    info['shared'].append((email, name))
-            del info["shared_with_name"]
-            del info["shared_with_email"]
+    sids = []
```

Comment:
This line is not needed, the code in 1374 should return an empty set if there are no studies at all.

Just tried something similar in the CLI and it works as expected:

```python
>>> x = set(i for i in [])
>>> x
set([])
```

Reply:
Thanks, in fact this is leftover from code transition ...